crates/key-server/src/server.rs

```diff
@@ -266,9 +307,14 @@ impl Server {
             vptb.ptb(),
             req_id
         );
+
+        // Add a staleness check as the first command in the PTB
+        let mut ptb = vptb.ptb().clone();
+        self.add_staleness_check_to_ptb(&mut ptb);
```

Comment:
below we should also catch the specific error code from our call and return the right error to the user